src/utils/index.ts

```diff
@@ -40,6 +40,7 @@ export function signedInfuraLink(
   return `https://${network}.infura.com/v3/${infura}`;
 }
 
+// Need more sophisticated way to determine if the contract is Legacy
```

Comment:
I think it's a pretty sophisticated way :) I'd probably just add another function that checks whether the contract is up-to-date (by checking its address), so we're sure that it has the interface we can rely on. If we say "it's not legacy, because it isn't a well-known legacy address", we likely delude ourselves, because we don't know, whether it's some kind of a custom resolver, about which we don't know anything, or it's our up-to-date resolver.

So I'd say, we need the following flow:
- Check if this is a well-known up-to-date resolver
  - If so, we are confident that we can call all the methods we need
  - Otherwise, we check it against other known legacy addresses
    - If it's a known legacy address, we are confident that we can call only the legacy set of methods
    - Otherwise, we need to make IO calls to actually check contract's interface
- Make the call, depending on the info we got in the previous step

Reply:
And yeah, what the current design is missing is compatibility with testnets.